ale_linters/crystal/ameba.vim

```diff
@@ -0,0 +1,56 @@
+" Author: Harrison Bachrach - https://github.com/HarrisonB
+" Description: Ameba, a linter for crystal files
+
+call ale#Set('crystal_ameba_executable', 'bin/ameba')
+
+function! ale_linters#crystal#ameba#GetCommand(buffer) abort
+    let l:executable = ale#Var(a:buffer, 'crystal_ameba_executable')
+
+    return ale#Escape(l:executable)
+    \   . ' --format json '
+    \   .  ale#Escape(expand('#' . a:buffer . ':p'))
+endfunction
+
+call ale#linter#Define('crystal', {
+\   'name': 'ameba',
+\   'executable_callback': ale#VarFunc('crystal_ameba_executable'),
+\   'command_callback': 'ale_linters#crystal#ameba#GetCommand',
+\   'callback': 'ale_linters#crystal#ameba#HandleAmebaOutput',
+\})
+
+" Handle output from ameba
+function! ale_linters#crystal#ameba#HandleAmebaOutput(buffer, lines) abort
+    try
+        let l:errors = json_decode(a:lines[0])
+    catch
+        return []
+    endtry
+
+    if !has_key(l:errors, 'summary')
+    \|| l:errors['summary']['issues_count'] == 0
+    \|| empty(l:errors['sources'])
+        return []
+    endif
+
+    let l:output = []
+
+    for l:error in l:errors['sources'][0]['issues']
+        let l:start_col = l:error['location']['column'] + 0
```

Comment:
why + 0? type coersion? i'd prefer str2nr() just because it's more obvious what it's doing

Reply:
from looking below at the tests it looks like it should be a number